### ISSUE TITLE
Fix S1144: rule should not report false positives with constants

### ISSUE THREAD

**Issue by tamasvajk** [SonarSource/sonar-csharp#904]
In the below example, the `Sample.Constants` class is faded out by S1144.

```C#
using System;

class Program
{
    static void Main(string[] args)
    {
        Console.WriteLine(Sample.Constants.X);
    }
}

public static class Sample
{
    internal static class Constants // Class reported as unused
    {
        public const string X = nameof(X);
    }
}
```

**Comment by Evangelink** [comment]
Hi @tamasvajk!

Thanks for the feedback! This rule is such a nightmare xD
